packages/calcite-components/src/components/table-row/table-row.tsx

```diff
@@ -158,7 +165,14 @@ export class TableRow extends LitElement implements InteractiveComponent {
     }
 
     if (changes.has("selected")) {
-      this.calciteInternalTableRowSelect.emit();
+      if (this.userTriggered) {
```

Comment:
Would two sequential `selected` updates *1 internal, 1 user-triggered) before `willUpdate` is called cause any issues with the component’s internal state?

Reply:
Good catch. Though not very common, i think in this case the first update is replaced and willUpdate when invoked considers the latest update on the property. This can cause issues with internal state of table me thinks.

Reply:
Updated to wait for update before emitting event.